noir_stdlib/src/meta/ctstring.nr

```diff
@@ -0,0 +1,98 @@
+use crate::append::Append;
+
+impl CtString {
+    // docs:start:new
+    comptime fn new() -> Self {
+        // docs::end::new
+        "".as_ctstring()
+    }
+
+    // Bug: using &mut self as the object results in this method not being found
+    // docs:start:append_str
+    comptime fn append_str<let N: u32>(self, s: str<N>) -> Self {
+        // docs:end:append_str
+        f"{self}{s}".as_ctstring()
+    }
+
+    // docs:start:append_fmtstr
+    comptime fn append_fmtstr<let N: u32, T>(self, s: fmtstr<N, T>) -> Self {
+        // docs:end:append_fmtstr
+        f"{self}{s}".as_ctstring()
+    }
+
+    /// CtString cannot directly return a str since the size would not be known.
+    /// To get around this, we return a quoted str and the underlying str can
+    /// be accessed using macro insertion `foo.as_quoted_str!()`.
+    // docs:start:as_quoted_str
+    comptime fn as_quoted_str(self) -> Quoted {
```

Comment:
To use this, would a user also need the length of the `CtString`? Please provide an example and/or test that uses this method

Reply:
There's already an example, `as_quoted_str_example` and it is in the docs as well.